r/src/array.cpp

```diff
@@ -516,6 +520,7 @@ inline SEXP simple_Array_to_Vector(const std::shared_ptr<arrow::Array>& array) {
 
   // first copy all the data
   auto p_values = GetValuesSafely<value_type>(array->data(), 1, array->offset());
+  STOP_IF_NULL(p_values);
```

Comment:
We could add an option to `GetValuesSafely` to abort if the pointer is null, would save some code duplication in this file

Reply:
But if we want to be able to write the error message here, e.g. `STOP_IF_NULL("invalid buffer because ....")` it's not that bad to have a separate thing